src/Util/Escape.php

```diff
@@ -116,18 +116,23 @@ public static function linuxArg($arg)
      */
     public static function windowsArg($arg)
     {
+        if ('' === $arg || null === $arg) {
+          return '""';
+        }
+        if (false !== strpos($arg, "\0")) {
+          $arg = str_replace("\0", '?', $arg);
+        }
+        if (!preg_match('/[\/()%!^"<>&|\s]/', $arg)) {
+          return $arg;
+        }
         // Double up existing backslashes
-        $arg = preg_replace('/\\\/', '\\\\\\\\', $arg);
-
-        // Double up double quotes
-        $arg = preg_replace('/"/', '""', $arg);
-
-        // Double up percents.
-        // $arg = preg_replace('/%/', '%%', $arg);
+        $arg = preg_replace('/(\\\\+)$/', '$1$1', $arg);
 
         // Replacing whitespace for good measure (see comment above).
         $arg = str_replace(["\t", "\n", "\r", "\0", "\x0B"], ' ', $arg);
 
+        $arg = str_replace(array('"', '^', '%', '!'), array('""', '"^^"', '"^%"', '"^!"'), $arg);
```

Comment:
Every version of php we support allows short-array syntax, so let's stick with that.

Reply:
Done